test/include/seqan3/test/pretty_printing.hpp

```diff
@@ -0,0 +1,32 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Contains an hacky overload for the googletest PrintTo function.
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ */
+
+#pragma once
+
+
+namespace seqan3
+{
+
+//!\cond DEV
+//!\brief Hacky overload for the googletest PrintTo function that always delegates to our debug_stream.
+template <typename t>
+    requires true
```

Comment:
Why this "constraint"?

Reply:
Okay – that's a clever way of overloading. And it works even if this function lives in seqan3 namespace?

Reply:
I'm not that familiar with the look-up rules, but [in the docs of googletest](https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#teaching-googletest-how-to-print-your-values) they also write that it needs to be in the same namespace as the type you want to print. (In the seconds snippet where they declare `printTo` as global function).

Though I think it only applies if you specialise it for a single type.
It will actually find both `printTo` functions (in `googletest` and `seqan3` namespace) and complain about the ambiguity if you don't use the constraint.

Reply:
Thanks! Then let's just add a (non-doxy) comment that this line must not be removed. Otherwise, someone might wonder about the same questions again.